File: scripts/patch-freight-delivery-field.js
const fs = require("fs");
const path = require("path");

const filePath = path.join(
  __dirname,
  "../src/app/receiving-management/page.jsx"
);
const lines = fs.readFileSync(filePath, "utf8").split(/\r?\n/);

if (lines.some((l) => l.includes('placeholder="Total freight for this PR"'))) {
  console.log("Already has freight field");
  process.exit(0);
}

const block = [
  '                  <motion.div className="col-md-6 mb-3">',
  '                    <label className="form-label">Freight cost</label>',
  "                    <input",
  '                      type="number"',
  '                      step="0.01"',
  '                      min="0"',
  '                      className="form-control"',
  '                      value={formData.freightCost ?? ""}',
  "                      onChange={(e) => handleFreightCostChange(e.target.value)}",
  '                      placeholder="Total freight for this PR"',
  "                    />",
  "                  </motion.div>",
];

const elOpen = "<" + "div";
const elClose = "</" + "motion.div>";
const fixed = block.map((line) =>
  line
    .replace("<motion.div", elOpen)
    .replace("</motion.div>", "</" + "motion.div>".replace("motion.", ""))
);

lines.splice(5706, 0, ...fixed);
fs.writeFileSync(filePath, lines.join("\n"));
console.log("Inserted freight field");

File: scripts/patch-freight-delivery-field.py
from pathlib import Path

p = Path(__file__).resolve().parent.parent / "src/app/receiving-management/page.jsx"
lines = p.read_text(encoding="utf-8").splitlines()
if any("Total freight for this PR" in line for line in lines):
    print("already")
    raise SystemExit(0)

d = "div"
block = [
    f'                  <{d} className="col-md-6 mb-3">',
    '                    <label className="form-label">Freight cost</label>',
    "                    <input",
    '                      type="number"',
    '                      step="0.01"',
    '                      min="0"',
    '                      className="form-control"',
    '                      value={formData.freightCost ?? ""}',
    "                      onChange={(e) => handleFreightCostChange(e.target.value)}",
    '                      placeholder="Total freight for this PR"',
    "                    />",
    f"                  </{d}>",
]
lines[5706:5706] = block
p.write_text("\n".join(lines) + "\n", encoding="utf-8")
print("ok")
